### PR TITLE
feat(docs): Add docs for ignite boilerplate files

### PR DESCRIPTION
**Pull request by Jpoliachik**
- Updated Boilerplate.md to reflect current project structure
- Added docs for several files / directories. Some may need more detail, but open to opinions!
- Updated ordering to reflect alphabetical filesystem, for consistency (open to opinions here)

A few files that are included in a new project, but we don't currently have docs for:
- babel.config.js
- jest.config.js
- metro.config.js
- react-native.config.js
- tsconfig.json 
- package.json
- README
Open to opinions on whether we want to exhaustively document every file or not. (do any of these files contain customizations that are worth documenting?)

### CODE REVIEW

File: docs/boilerplate/app/devtools/Devtools.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Devtools Folder
+
+## Reactotron
+
+Ignite comes with Reactotron support for debugging your app.
+By default, Reactotron is configured to work with web and mobile apps and is configured with a few plugins and commands we think are useful.
+
+### ReactotronConfig.ts
+
+The `reactotron-mst` plugin is included for Mobx-State-Tree support.
```
Comment (frankcalise) [inline docs/boilerplate/app/devtools/Devtools.md:10]: nit: Mobx -> MobX

:D

File: docs/boilerplate/plugins/Plugins.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+```typescript
+// In app.config.ts
+plugins: [...existingPlugins, require("./plugins/yourCustomPlugin").yourCustomPlugin]
+```
+
+## Key Points
+
+- Config plugins extend app configuration, automating native module integration.
+- Create plugins in app/plugins and add them to app.config.ts.
```
Comment (frankcalise) [inline docs/boilerplate/plugins/Plugins.md:22]: nit: backtick around `app/plugins` and `app.config.ts`

**Review by frankcalise — APPROVED** [review]
Thanks for adding some nice content here! Just a couple of nits and I think it's good to go

File: docs/boilerplate/app/devtools/Devtools.md (hunk 1 above)
Comment (Jpoliachik) [reply docs/boilerplate/app/devtools/Devtools.md:10]: lol I need to drill this into memory

**Pull request merged**